src/dda/utils/git/changeset.py

```diff
@@ -159,6 +159,12 @@ def __or__(self, other: Self) -> Self:
     def __eq__(self, other: object) -> bool:
         return isinstance(other, ChangeSet) and self.paths == other.paths
 
+    def __len__(self) -> int:
+        return len(self.__files)
+
+    def __bool__(self) -> bool:
+        return bool(self.__files)
```

Comment:
Let's just have callers access the `files` attribute instead so we don't have to think about what is or is not supported at the class level.